### DISCUSSION TITLE
Does Blacksheep support built-in multiple hosting ?

### DISCUSSION THREAD

**Discussion by i-Ching** [Neoteroi/BlackSheep#292] [Q&A] [answered]
Now the question is Blacksheep turn:

[https://github.com/starlite-api/starlite/discussions/939](https://github.com/starlite-api/starlite/discussions/939)

I have gotten a clear answer from Starlite.
Having read your official website, I cannot find any hint.

Does Blacksheep support built-in multiple hosting ? or future ?

**Comment by RobertoPrevato** [comment, marked as answer]
Hi @i-Ching 
Thank You for opening this discussion. Currently Blacksheep does not support this, but I like the idea of adding support for it, and I can implement support for this.

Currently I am working on version 2 of the framework and related libraries. Right now adding support for dependency injection to authentication handlers and authorization requirements - I also plan to add full support for dependency injection to binders.
I can add support for routing based on HTTP headers (including host) and make it part of v2.
